### PR TITLE
Run null check on needed API keys before loading cogs.

### PR DESCRIPTION
**Pull request by shtlrs**
We have a lot of Cogs that need some API key to function.

If the key isn't present, it would result in constant 401s.

This PR adds a change that won't load these Cogs & log a warning whenever these keys aren't present.

### CODE REVIEW

**Review by ChrisLovering — APPROVED** [review]

File: bot/exts/fun/snakes/__init__.py [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
+from bot.constants import Tokens
 from bot.exts.fun.snakes._snakes_cog import Snakes
 
 log = logging.getLogger(__name__)
 
 
 async def setup(bot: Bot) -> None:
     """Load the Snakes Cog."""
+    if not Tokens.giphy:
+        log.warning("No Youtube token. All youtube related commands in Snakes cog won't work.")
```
Comment (wookie184) [inline bot/exts/fun/snakes/__init__.py:13]: Looks like the wrong constant? Also capitalise youtube as `YouTube`

File: bot/exts/utilities/wolfram.py [hunk 2]
```diff
@@ -303,4 +303,7 @@ async def wolfram_short_command(self, ctx: Context, *, query: str) -> None:
 
 async def setup(bot: Bot) -> None:
     """Load the Wolfram cog."""
+    if not Wolfram.key:
```
Comment (wookie184) [inline bot/exts/utilities/wolfram.py:306]: `Wolfram` is the cog, not the constants config. This causes an error.

**Review by wookie184 — CHANGES_REQUESTED** [review]
There's currently an error log for `bot.exts.utilities.reddit` when missing the token, could we make that a warning for consistency?

File: bot/exts/fun/snakes/__init__.py (hunk 1 above)
Comment (shtlrs) [reply bot/exts/fun/snakes/__init__.py]: 5bdd195aefc1b3af562e336dd9edeb260002e085

Comment (shtlrs) [reply bot/exts/fun/snakes/__init__.py]: > There's currently an error log for `bot.exts.utilities.reddit` when missing the token, could we make that a warning for consistency?

Yessir we can 0e202c04bd6c5976afdcd4d10a62fde386dcf440

File: bot/exts/utilities/wolfram.py (hunk 2 above)
Comment (shtlrs) [reply bot/exts/utilities/wolfram.py]: My bad, I hadn't still tested this yet, and I was trying to get something in a PR before I left the house.
5ba5275ab343ec36f2a2ab99f7d5a6dd2f4db8f6

File: bot/exts/holidays/halloween/spookygif.py [hunk 3]
```diff
@@ -35,4 +35,7 @@ async def spookygif(self, ctx: commands.Context) -> None:
 
 async def setup(bot: Bot) -> None:
     """Spooky GIF Cog load."""
+    if not Tokens.giphy:
+        log.warning("No Giphy token. Not loading Giphy cog.")
```
Comment (wookie184) [inline bot/exts/holidays/halloween/spookygif.py:39]:
```suggestion
        log.warning("No Giphy token. Not loading SpookyGif cog.")
```

**Review by wookie184 — CHANGES_REQUESTED** [review]
looks good, missed one thing on my previous review.

**Review by wookie184 — APPROVED** [review]
This is a nice QOL improvement, thanks!

**Pull request merged**